custom_components/xiaomi_home/miot/miot_lan.py

```diff
@@ -714,76 +700,99 @@ def unsub_lan_state(self, key: str) -> None:
 
     @final
     def sub_device_state(
-        self, key: str, handler: Callable[[str, dict, any], None],
-        handler_ctx: any = None
+        self, key: str, handler: Callable[[str, dict, Any], Coroutine],
+        handler_ctx: Any = None
     ) -> bool:
-        return self.__lan_send_cmd(
-            cmd=MIoTLanCmdType.SUB_DEVICE_STATE,
-            data=MIoTLanSubDeviceState(
+        self._internal_loop.call_soon_threadsafe(
+            self.__sub_device_state,
+            _MIoTLanSubDeviceData(
                 key=key, handler=handler, handler_ctx=handler_ctx))
+        return True
 
     @final
     def unsub_device_state(self, key: str) -> bool:
-        return self.__lan_send_cmd(
-            cmd=MIoTLanCmdType.UNSUB_DEVICE_STATE,
-            data=MIoTLanUnsubDeviceState(key=key))
+        self._internal_loop.call_soon_threadsafe(
+            self.__unsub_device_state, _MIoTLanUnsubDeviceData(key=key))
+        return True
 
     @final
     def sub_prop(
-        self, did: str, handler: Callable[[dict, any], None],
-        siid: int = None, piid: int = None, handler_ctx: any = None
+        self,
+        did: str,
+        handler: Callable[[dict, Any], None],
+        siid: Optional[int] = None,
+        piid: Optional[int] = None,
+        handler_ctx: Any = None
     ) -> bool:
         if not self._enable_subscribe:
             return False
         key = (
             f'{did}/p/'
             f'{"#" if siid is None or piid is None else f"{siid}/{piid}"}')
-        return self.__lan_send_cmd(
-            cmd=MIoTLanCmdType.REG_BROADCAST,
-            data=MIoTLanRegisterBroadcastData(
+        self._internal_loop.call_soon_threadsafe(
+            self.__sub_broadcast,
+            _MIoTLanRegisterBroadcastData(
                 key=key, handler=handler, handler_ctx=handler_ctx))
+        return True
 
     @final
-    def unsub_prop(self, did: str, siid: int = None, piid: int = None) -> bool:
+    def unsub_prop(
+        self,
+        did: str,
+        siid: Optional[int] = None,
+        piid: Optional[int] = None
+    ) -> bool:
         if not self._enable_subscribe:
             return False
         key = (
             f'{did}/p/'
             f'{"#" if siid is None or piid is None else f"{siid}/{piid}"}')
-        return self.__lan_send_cmd(
-            cmd=MIoTLanCmdType.UNREG_BROADCAST,
-            data=MIoTLanUnregisterBroadcastData(key=key))
+        self._internal_loop.call_soon_threadsafe(
```

Comment:
After deinit_async called, the program will stop internal loop and call the lan state change callback. When other services judge that the lan state changed, functions such as unsub_prop may be called, which may lead to RuntimeError.

Reply:
Thanks for the catch. I'll add a readiness assertion to avoid access violations. Which should throw a specific exception if the service is not ready.

Reply:
Fixed in 0edb526e8ebfa994b8570d4981f0c523c28ec152